### PR TITLE
Queen piece class with test

### PR DESCRIPTION
**Pull request by Rmunitz-oss**

### CODE REVIEW

**Review by wolfishl — APPROVED** [review]
Looks great! It's going to need the extra boolean for move when the captureOnly flag is merged

**Review by scroitoru — APPROVED** [review]

File: src/main/java/touro/chess/QueenPiece.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+public class QueenPiece extends AbstractPiece{
+
+    public QueenPiece(Location location, PieceColor color) {
+        super(location, color);
+    }
+
+    @Override
+    public List<Move> getMoves() {
+        List<Move> moveList = new ArrayList<>();
+        Location currentLocation = this.getLocation();
```
Comment (andrewoid) [inline src/main/java/touro/chess/QueenPiece.java:15]: `this.` is not necessary.

File: src/main/java/touro/chess/QueenPiece.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        int row = currentLocation.getRow();
+        int col = currentLocation.getColumn();
+        int newCol;
+        int newRow;
+
+        //horizontal moves:
+        for (newCol = 1; newCol <= 8; newCol++){
```
Comment (andrewoid) [inline src/main/java/touro/chess/QueenPiece.java:22]: `newCol` should start from 0 not 1.

File: src/test/java/touro/chess/QueenPieceTest.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
+package touro.chess;
+
+import org.junit.Test;
+import java.util.ArrayList;
+import java.util.List;
+import static org.junit.Assert.assertEquals;
+
+public class QueenPieceTest {
+    @Test
+    public void testGetMoves() {
```
Comment (andrewoid) [inline src/test/java/touro/chess/QueenPieceTest.java:10]: just `getMoves()` 
You can drop the "test" from the name of the method.

File: src/main/java/touro/chess/Square.java [hunk 4]
```diff
@@ -1,5 +1,5 @@
 package touro.chess;
-
+/**
```
Comment (andrewoid) [inline src/main/java/touro/chess/Square.java:2]: Why have you done this?

Comment (Rmunitz-oss) [reply src/main/java/touro/chess/Square.java:2]: Oh I didn't mean to push that! For some reason was having issues with this import: `import org.jetbrains.annotations.Nullable;` It was easier to comment out that class for now so I could run my tests. 
Should I uncomment and push that class again?

Comment (andrewoid) [reply src/main/java/touro/chess/Square.java:2]: Yes. You cannot comment out code just because it doesn't work for you.

**Pull request merged**